datajunction-server/datajunction_server/models/node.py

```diff
@@ -234,8 +282,74 @@ class AvailabilityStateBase(BaseSQLModel):
     schema_: Optional[str] = Field(default=None)
     table: str
     valid_through_ts: int
-    max_partition: List[str] = Field(sa_column=SqlaColumn(JSON))
-    min_partition: List[str] = Field(sa_column=SqlaColumn(JSON))
+
+    # Node-level temporal ranges
+    min_partition: Optional[List[str]] = Field(sa_column=SqlaColumn(JSON))
+    max_partition: Optional[List[str]] = Field(sa_column=SqlaColumn(JSON))
+
+    # Specific partitions and their temporal ranges.
+    partitions: Optional[List[AvailablePartition]] = Field(
```

Comment:
I think you are assuming that any given node can only have a single non-temporal partition key. Is that correct or am I missing something?